frontend/src/modules/alarms/AlarmsPage.tsx

```diff
@@ -292,41 +294,71 @@ export const AlarmsPage = (): JSX.Element => {
     setPage(value);
   };
 
-  const alarmLimitsRequestStandby = useSelector(getAlarmLimitsRequestStandby);
+  const alarmLimitsRequestDraft = useSelector(getAlarmLimitsRequestDraft);
+  const alarmLimitsCurrent = useSelector(getAlarmLimitsCurrent);
   const dispatch = useDispatch();
   const currentMode = useSelector(getParametersRequestMode);
   const ventilating = useSelector(getParametersIsVentilating);
-  const setAlarmLimits = (data: Partial<AlarmLimitsRequest>) => {
+  const alarmLimitsUnsaved = useSelector(getAlarmLimitsUnsavedChanges);
+  const [alarmLimits] = useState((alarmLimitsCurrent as unknown) as Record<string, Range>);
+  const alarmLimitsDraft = (alarmLimitsRequestDraft as unknown) as Record<string, Range>;
+  const setAlarmLimitsRequestDraft = (data: Partial<AlarmLimitsRequest>) => {
     dispatch(commitStandbyRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, data));
   };
   const applyChanges = () => {
-    if (alarmLimitsRequestStandby === null) {
+    if (alarmLimitsRequestDraft === null) {
       return;
     }
 
     dispatch(
-      commitRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, alarmLimitsRequestStandby),
+      commitRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, alarmLimitsRequestDraft),
     );
   };
   const alarmConfig = alarmConfiguration(currentMode);
   const [open, setOpen] = useState(false);
+  const [discardOpen, setDiscardOpen] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(!alarmLimitsUnsaved);
 
   const handleClose = () => {
     setOpen(false);
   };
 
+  const handleDiscardClose = () => {
+    setDiscardOpen(false);
+  };
+
   const handleConfirm = () => {
     setOpen(false);
     applyChanges();
+    setIsDisabled(true);
+  };
+
+  const handleDiscardConfirm = () => {
+    setDiscardOpen(false);
+    setAlarmLimitsRequestDraft(alarmLimits);
+    setIsDisabled(true);
   };
+
   const handleOpen = () => {
```

Comment:
Does this actually mean `handleConfirmOpen`?

Reply:
Done.